### PR TITLE
Add changelog for 2023.7 and bump version in pyproject.toml

### PR DESCRIPTION
**Pull request by hugovk**
Preperation for a new release.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,19 +2,35 @@
...
+   Contributed by Hugo van Kemenade
+- Test 3.12 branch, now `main` is 3.13 (#140)
+   Contributed by Hugo van Kemenade
+
 `2023.5 <https://github.com/python/python-docs-theme/releases/tag/v2023.5>`_
 ----------------------------------------------------------------------------
 
 - Add a dark theme. (#44)
-  Contributed by Nils K
+   Contributed by Nils K
```
Comment (pradyunsg) [inline CHANGELOG.rst:25]: Unrelated changes?

**Review by pradyunsg — APPROVED** [review]
LGTM, with 1 non-blocking concern.

File: CHANGELOG.rst (hunk 1 above)
Comment (hugovk) [reply CHANGELOG.rst:25]: Sure, the early releases were done in this format:

* https://github.com/python/python-docs-theme/blob/main/CHANGELOG.rst#20215
* https://github.com/python/python-docs-theme/blob/main/CHANGELOG.rst#202012

But not the later. It seems reasonable to use the same format for later releases, it's also more readable.

I can put it in another PR if you prefer, or revert.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,19 +2,35 @@
...
 =========
 
+`2023.7 <https://github.com/python/python-docs-theme/releases/tag/v2023.7>`_
+----------------------------------------------------------------------------
+
+- Do not append asset hashes on Sphinx 7.1+ (#137)
+   Contributed by Pradyun Gedam
+- Use correct block, enable built-in slash keypress (#141)
+   Contributed by Mike Fiedler
+- Sphinx 6.2 fix: add ``nav.contents`` where that ``div.topic`` is used (#138)
```
Comment (AA-Turner) [inline CHANGELOG.rst:12]: Grammar
```suggestion
- Sphinx 6.2 fix: add ``nav.contents`` where ``div.topic`` is used (#138)
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,19 +2,35 @@
...
+   Contributed by Hugo van Kemenade
+- Dark mode: fix contrast for C++ specific styling (#133)
```
Comment (AA-Turner) [inline CHANGELOG.rst:14]:
```suggestion
- Dark mode: fix contrast for ``C++`` specific styling (#133)
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -2,19 +2,35 @@
...
+  Contributed by Hugo van Kemenade
+- Don't let long code literals extend beyond the right side of the screen (#139)
+   Contributed by Hugo van Kemenade
+- Test 3.12 branch, now `main` is 3.13 (#140)
```
Comment (AA-Turner) [inline CHANGELOG.rst:18]: More descriptive ('branch' implicitly refers to CPython here, but it isn't very clear)
```suggestion
- Test with Python 3.12 (#140)
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -2,19 +2,35 @@
 Changelog
 =========
 
+`2023.7 <https://github.com/python/python-docs-theme/releases/tag/v2023.7>`_
+----------------------------------------------------------------------------
+
+- Do not append asset hashes on Sphinx 7.1+ (#137)
```
Comment (AA-Turner) [inline CHANGELOG.rst:8]: This suggests a potential regression, that P-D-T no longer does the hashes -- whereas the functionality was added to Sphinx core instead.
```suggestion
- Fix compatability with Sphinx 7.1 (#137)
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -2,19 +2,35 @@
...
+   Contributed by Pradyun Gedam
+- Use correct block, enable built-in slash keypress (#141)
```
Comment (AA-Turner) [inline CHANGELOG.rst:10]: Clarity on the user-facing change:
```suggestion
- Enable the slash keypress to focus the search field (#141)
```

**Review by AA-Turner — COMMENTED** [review]
Some suggestions on re-wording to make changes clearer to users of the theme:

A

File: CHANGELOG.rst (hunk 3 above)
Comment (hugovk) [reply CHANGELOG.rst:14]: Thanks! I added all suggestions except this as we don't need code literals for proper nouns.

**Review by AA-Turner — APPROVED** [review]

**Comment by hugovk** [comment]
@Mariatta Please could you release 2023.7 to PyPI? There are instructions at https://github.com/python/python-docs-theme/blob/main/CONTRIBUTING.rst and the first two steps have been done here. Thank you!

**Comment by Mariatta** [comment]
@hugovk just released v 2023.7. Sorry for the delay.
